pankti-extract/src/main/java/se/kth/castor/pankti/extract/util/MethodUtil.java

```diff
@@ -136,4 +137,16 @@ private static boolean isNestedInvocationMockable(final CtMethod<?> method,
         }
         return nestedMethodInvocationMap;
     }
+
+    /**
+     * Checks if the declaring type of a method has a default constructor.
+     *
+     * @param method The method
+     * @return true if the declaring type has a default constructor
+     */
+    public static boolean declaringTypeHasDefaultConstructor(final CtMethod<?> method) {
+        return method.getDeclaringType()
+                .getElements(new TypeFilter<>(CtConstructor.class))
+                .stream().anyMatch(c -> c.getParameters().isEmpty());
+    }
```

Comment:
There's either a documentation error or an implementation error here. A constructor without parameters is not necessarily the default constructor. The default constructor is specifically the constructor that the compiler creates by default if there's no declared constructor ([definition in the JLS](https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.8.9)).

What I _think_ you're looking for here is just any parameterless constructor, default or not, as that makes creating a mock very easy. If that's the case I'd suggest updating the docs and method name to reflect this.

If you're _actually_ looking for the default constructor, then the way to find it in Spoon is to check for an implicit constructor. I'd also advise adding a test case to ensure that you don't pick up a non-default parameterless constructor.

```suggestion
    /**
     * Checks if the declaring type of a method has a default constructor.
     *
     * @param method The method
     * @return true if the declaring type has a default constructor
     */
    public static boolean declaringTypeHasDefaultConstructor(final CtMethod<?> method) {
        return method.getDeclaringType()
                .getElements(new TypeFilter<>(CtConstructor.class))
                .stream().anyMatch(CtConstructor::isImplicit);
    }
```

Reply:
Yes, you're right. I'm actually looking for non-parameterized constructors. I'll change the name to reflect this! Thanks!